src/main/java/org/jboss/pnc/deliverablesanalyzer/CacheProvider.java

```diff
@@ -135,11 +201,51 @@ public DefaultCacheManager initCaches() throws IOException {
         return cacheManager;
     }
 
-    public void close(@Disposes DefaultCacheManager cacheManager) {
-        try {
-            cacheManager.close();
-        } catch (IOException e) {
-            LOGGER.warn("Failed to close cache manager {}", cacheManager.getCache().getName(), e);
+    /**
+     * Setup the distributed Infinispan cache manager from configs. It is assumed that the caches are already setup on
+     * the remote Infinispan server.
+     *
+     * In the future, this can be replaced by the built-in quarkus-infinispan-client once we don't use the embedded
+     * infinispan feature.
+     *
+     * @return remote cache manager setup to talk to the distributed infinispan server
+     * @throws RuntimeException If the infinispan remote server list, username, or password are not specified in the
+     *         config
+     */
+    private RemoteCacheManager setupDistributedCacheManager() {
+
+        throwRuntimeExceptionIfOptionalEmpty(infinispanRemoteServerList, "infinispan server list");
+        throwRuntimeExceptionIfOptionalEmpty(infinispanUsername, "infinispan username");
+        throwRuntimeExceptionIfOptionalEmpty(infinispanPassword, "infinispan password");
+
+        ConfigurationBuilder builder = new ConfigurationBuilder();
+
+        for (String server : infinispanRemoteServerList.get()) {
```

Comment:
Several `get`s here that weren't checked for `isPresent`.

Reply:
They are checked prior at the start of the method.

Reply:
Yup they could be unset, and maybe the empty string is a valid value. Who knows.

Reply:
From Infinispan server 11.0.3.Final server zip:
```
~/infinispan/infinispan-server-11.0.3.Final/bin ❯ ./cli.sh user create test          
Set a password for the user: 
Set a password for the user: 
Set a password for the user: 
Set a password for the user: 
```

You cannot set an empty password for a user. A null username also doesn't make sense because you have to create at least a user to talk to Infinispan. So I think an empty password is an unrealistic scenario.

Reply:
OK.